go-controller/pkg/node/base_node_network_controller_dpu.go

```diff
@@ -14,139 +13,214 @@ import (
 	"k8s.io/client-go/util/retry"
 	"k8s.io/klog/v2"
 
+	nettypes "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cni"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
 )
 
+// Check if the Pod is ready so that we can add its associated DPU to br-int.
+// If true, return its dpuConnDetails, otherwise return nil
+func (bnnc *BaseNodeNetworkController) podReadyToAddDPU(pod *kapi.Pod, nadName string) *util.DPUConnectionDetails {
+	if bnnc.name != pod.Spec.NodeName {
+		klog.V(5).Infof("Pod %s/%s is not scheduled on this node %s", pod.Namespace, pod.Name, bnnc.name)
+		return nil
+	}
+
+	dpuCD, err := util.UnmarshalPodDPUConnDetails(pod.Annotations, nadName)
+	if err != nil {
+		if !util.IsAnnotationNotSetError(err) {
+			klog.Errorf("Failed to get dpu annotation for pod %s/%s nad %s: %v",
```

Comment:
DPU and NAD

Reply:
@trozet thank you for all your comments. Fixed.